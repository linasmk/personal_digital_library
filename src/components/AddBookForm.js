/* ============ Dependencies ============= */
import React from "react";
import Modal from "react-modal";
import { AiOutlineClose} from "react-icons/ai";
/* ============== Redux =============== */

/* ============ Components ============== */
/* =============== Code ================= */
//Modal.setAppElement("#app");
class AddBookForm extends React.Component {
  state = {
    title: this.props.book ? this.props.book.title : "",
    author: this.props.book ? this.props.book.author : "",
    description: this.props.book ? this.props.book.description : "",
    publicationYear: this.props.book ? this.props.book.publicationYear : "",
    publisher: this.props.book ? this.props.book.publisher : "",
    publicationLength: this.props.book ? this.props.book.publicationLength : "",
    category: this.props.book ? this.props.book.category : "",
    url: this.props.book ? this.props.book.url : "",
    rating: this.props.book ? this.props.book.rating : "",
    imgLink: this.props.book ? this.props.book.imgLink : "",
    error: "",
  };
  onTitleChange = (e) => {
    const title = e.target.value;
    this.setState(() => ({
      title,
    }));
  };
  onAuthorChange = (e) => {
    const author = e.target.value;
    this.setState(() => ({
      author,
    }));
  };
  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({
      description,
    }));
  };
  onPublicationYearChange = (e) => {
    const publicationYear = e.target.value;
    this.setState(() => ({
      publicationYear,
    }));
  };

  onPublicationLengthChange = (e) => {
    const publicationLength = e.target.value;
    this.setState(() => ({
      publicationLength,
    }));
  };

  onPublisherChange = (e) => {
    const publisher = e.target.value;
    this.setState(() => ({
      publisher,
    }));
  };

  onUrlChange = (e) => {
    const url = e.target.value;
    this.setState(() => ({
      url,
    }));
  };
  
  onCategoryChange = (e) => {
    const category = e.target.value;
    this.setState(() => ({
      category,
    }));
  };
  
  onRatingChange = (e) => {
    const rating = e.target.value;
    this.setState(() => ({
      rating,
    }));
  };
  onImgLinkChange = (e) => {
    const imgLink = e.target.value;
    this.setState(() => ({
      imgLink,
    }));
  };
  onSubmit = (e) => {
    e.preventDefault();
    if (!this.state.title) {
      this.setState(() => ({
        error: "Title's name field should not stay empty!",
      }));
    } else {
      this.setState(() => ({
        error: "",
      }));
      this.props.onSubmit({
        title: this.state.title,
        author: this.state.author,
        description: this.state.description,
        publicationYear: this.state.publicationYear,
        publisher: this.state.publisher,
        publicationLength: this.state.publicationLength,
        category: this.state.category,
        url: this.state.url,
        rating: this.state.rating,
        imgLink: this.state.imgLink,
      });
      this.props.clearAddNewBookModal();
    }
  };
  render() {
    return (
      <Modal
        isOpen={this.props.addNewBookModal}
        onRequestClose={this.props.clearAddNewBookModal}
        contentLabel={"Add New Book"}
        closeTimeoutMS={800}
        className="modal"
        ariaHideApp={false}
      >
      <div className="top-bar h-top-bar-modal">
        <AiOutlineClose
            className="card__btn-close h-icon-styles"
            viewBox="100 50 800 880"
            onClick={this.props.clearAddNewBookModal}
          />
      </div>
        <h2 className="addbook__heading">Add New Book</h2>
        <form className="addbook__form" onSubmit={this.onSubmit}>
          {this.state.error && (
            <p className="addbook-form__warning">{this.state.error}</p>
          )}
          <div className="addbook__form_row">
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="text"
                required
                autoComplete="off"
                value={this.state.title}
                onChange={this.onTitleChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">Book Title</span>
              </label>
            </div>
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="text"
                required
                autoComplete="off"
                value={this.state.author}
                onChange={this.onAuthorChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">Author</span>
              </label>
            </div>
          </div>
          <div className="addbook__form_row">
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="number"
                required
                value={this.state.publicationYear}
                onChange={this.onPublicationYearChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">Publication Year</span>
              </label>
            </div>
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="number"
                required
                value={this.state.publicationLength}
                onChange={this.onPublicationLengthChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">
                  Publication Length
                </span>
              </label>
            </div>
          </div>

          <div className="addbook__form_row">
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="text"
                required
                value={this.state.publisher}
                onChange={this.onPublisherChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">Publisher</span>
              </label>
            </div>
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="url"
                title="URL should match the pattern: https://.*"
                required
                pattern="https://.*"
                autoComplete="off"
                value={this.state.url}
                onChange={this.onUrlChange}
              />
              <label htmlFor="url" className="addbook__form_label">
                <span className="addbook__form_content">Link to the Book</span>
              </label>
            </div>
          </div>

          <div className="addbook__form_row">
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="text"
                required
                value={this.state.category}
                onChange={this.onCategoryChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">Category</span>
              </label>
            </div>

            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="number"
                min="1"
                max="10"
                required
                value={this.state.rating}
                onChange={this.onRatingChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">Rating</span>
              </label>
            </div>
          </div>
          <div className="addbook__form_row">
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="url"
                title="URL should match the pattern: https://.*"
                required
                autoComplete="on"
                value={this.state.imgLink}
                onChange={this.onImgLinkChange}
              />
              <label htmlFor="url" className="addbook__form_label">
                <span className="addbook__form_content">Image Link</span>
              </label>
            </div>
          </div>
          <div className="addbook__form_row">
            <div className="addbook__form_column-textarea">
              <textarea
                className="addbook__form_textarea"
                type="text"
                required
                maxLength="3000"
                value={this.state.description}
                onChange={this.onDescriptionChange}
              />
              <label htmlFor="title" className="addbook__form_label-textarea">
                <span className="addbook__form_content-textarea">
                  Book Description
                </span>
              </label>
            </div>
          </div>
          <div className="modal__button-bottom_wrapper">
            <button className="modal__button-bottom" type="submit">
              Add Book
            </button>
          </div>
        </form>
      </Modal>
    );
  }
}

export default AddBookForm;
