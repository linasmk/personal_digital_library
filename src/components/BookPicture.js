import React from "react";
const BookPicture = (props) => {
  return (
    <picture>
      <source media="(min-width: 1400px)" srcSet={props.srcXlg} sizes="250px" />
      <source media="(min-width: 998px)" srcSet={props.srcLg} sizes="200px" />
      <source media="(min-width: 480px)" srcSet={props.srcMd} sizes="120px" />
      <source media="(min-width: 200px)" srcSet={props.srcSm} sizes="80px" />
      <img className="book__picture" src={props.src} alt="title" />
    </picture>
  );
};
export default BookPicture;
