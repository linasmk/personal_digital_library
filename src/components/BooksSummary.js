/* ========= App Dependencies ============= */
import React from "react";
import { connect } from "react-redux";
/* ========== Redux ========== */
import getVisibleBooks from "../store/selectors";

/* ========= Code ============= */
export const BooksSummary = ({ booksCount }) => {
  return (
    <section className="header__summary_block">
      <p className="header__summary_p">
        Filtered books:{" "}
        <span className="header__summary_output">{booksCount}</span>
      </p>
    </section>
  );
};

const mapStateToProps = (state) => {
  const visibleBooks = getVisibleBooks(state.books, state.filters);
  return {
    booksCount: visibleBooks.length,
  };
};

export default connect(mapStateToProps)(BooksSummary);
