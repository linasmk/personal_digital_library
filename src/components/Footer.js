import React from "react";
const Footer = props => {
  const currentYear = new Date().getFullYear();
  
      
  
  return (
    <footer className="footer">
      <div className="footer__overlay">
        <h3 className="footer__copyright">
          &copy; {currentYear}. Linas Mackonis
        </h3>
      </div>
    </footer>
  );
};
export default Footer;
