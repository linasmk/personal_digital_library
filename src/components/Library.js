import React from "react";
const library = [
  {
    id: 2000,
    title:
      "Gamify: How Gamification Motivates People to Do Extraordinary Things",
    author: "Brian Burke",
    description:
      "Gamify shows gamification in action: as a powerful approach to engaging and motivating people to achieving their goals, while at the same time achieving organizational objectives. It can be used to motivate people to change behaviors, develop skills, and drive innovation. The key to gamification success is to engage people on an emotional level and motivating them to achieve their goals.",
    publicationYear: 2014,
    publicationLength: 195,
    category: "business",
    url:
      "https://www.amazon.com/Gamify-Gamification-Motivates-People-Extraordinary-ebook/dp/B00I2ZW47W",
    rating: 8,
    src: "img/book_covers/gamify324x500.jpg",
    srcSm: "img/book_covers/gamify80x123.jpg",
    srcMd: "img/book_covers/gamify120x185.jpg",
    srcLg: "img/book_covers/gamify200x307.jpg",
    srcXlg: "img/book_covers/gamify250x386.jpg",
    background:
      "linear-gradient(185deg, rgb(0, 255, 255), rgb(255, 255, 255), rgb(0, 255, 255));"
  },
  {
    id: 5000,
    title: "Mind As Machine: A History of Cognitive Science Two-Volume Set",
    author: "Margaret Boden",
    description:
      "Psychology is the thematic heart of cognitive science, which aims to understand human (and animal) minds. But its core theoretical ideas are drawn from cybernetics and artificial intelligence, and many cognitive scientists try to build functioning models of how the mind works. In that sense, Margaret Boden suggests, its key insight is that mind is a (very special) machine. Because the mind has many different aspects, the field is highly interdisciplinary. It integrates psychology not only with cybernetics/AI, but also with neuroscience and clinical neurology; with the philosophy of mind, language, and logic.",
    publicationYear: 2000,
    publicationLength: 1712,
    category: "cognitive science",
    url: "https://www.amazon.com/Mind-As-Machine-Cognitive-Two/dp/0199241449",
    rating: 8,
    src: "img/book_covers/mind_as_machine396x499.jpg",
    srcSm: "img/book_covers/mind_as_machine80x101.jpg",
    srcMd: "img/book_covers/mind_as_machine120x151.jpg",
    srcLg: "img/book_covers/mind_as_machine200x252.jpg",
    srcXlg: "img/book_covers/mind_as_machine250x315.jpg",
    background:
      "linear-gradient(185deg, rgb(0, 255, 255), rgb(255, 255, 255), rgb(0, 255, 255));"
  },
  {
    id: 5001,
    title: "Bottlenecks: Aligning UX Design with User Psychology",
    author: "David C. Evans",
    description:
      "This book fills a need for entrepreneurs, designers, and marketing professionals in the application of foundational psychology to user-experience design. The first generation of books on the topic focused on web pages and cognitive psychology. This book covers apps, social media, in-car infotainment, and multiplayer video games, and it explores the crucial roles played by behaviorism, development, personality, and social psychology.",
    publicationYear: 2017,
    publicationLength: 284,
    category: "cognitive science",
    url:
      "https://www.amazon.com/Bottlenecks-Aligning-Design-User-Psychology-ebook-dp-B06W2J32CK/dp/B06W2J32CK/ref=mt_kindle?_encoding=UTF8&me=&qid=",
    rating: 9,
    src: "img/book_covers/bottlenecks330x499.jpg",
    srcSm: "img/book_covers/bottlenecks80x120.jpg",
    srcMd: "img/book_covers/bottlenecks120x181.jpg",
    srcLg: "img/book_covers/bottlenecks200x302.jpg",
    srcXlg: "img/book_covers/bottlenecks250x379.jpg",
    background:
      "linear-gradient(185deg, rgb(0, 255, 255), rgb(255, 255, 255), rgb(0, 255, 255));"
  },
  {
    id: 6000,
    title: "JavaScript: The Good Parts",
    author: "Douglas Crockford",
    description:
      "Most programming languages contain good and bad parts, but JavaScript has more than its share of the bad, having been developed and released in a hurry before it could be refined. This authoritative book scrapes away these bad features to reveal a subset of JavaScript that's more reliable, readable, and maintainable than the language as a whole subset you can use to create truly extensible and efficient code.",
    publicationYear: 2008,
    publicationLength: 176,
    category: "programming languages",
    url:
      "https://www.amazon.com/JavaScript-Good-Parts-Douglas-Crockford/dp/0596517742",
    rating: 8,
    src: "img/book_covers/js_the_good_parts383x499.jpg",
    srcSm: "img/book_covers/js_the_good_parts80x104.jpg",
    srcMd: "img/book_covers/js_the_good_parts120x156.jpg",
    srcLg: "img/book_covers/js_the_good_parts200x260.jpg",
    srcXlg: "img/book_covers/js_the_good_parts250x325.jpg",
    background:
      "linear-gradient(185deg, rgb(0, 255, 255), rgb(255, 255, 255), rgb(0, 255, 255));"
  },
  {
    id: 1000,
    title: "The Power of Habit: Why We Do What We Do in Life and Business",
    author: "Charles Duhigg",
    description:
      "Duhigg presents a whole new understanding of human nature and its potential. At its core, The Power of Habit contains an exhilarating argument: The key to exercising regularly, losing weight, being more productive, and achieving success is understanding how habits work. As Duhigg shows, by harnessing this new science, we can transform our businesses, our communities, and our lives.",
    publicationYear: 2012,
    publicationLength: 400,
    category: "psychology",
    url:
      "https://www.amazon.com/Power-Habit-What-Life-Business/dp/1400069289/ref=tmm_hrd_swatch_0?_encoding=UTF8&qid=1582294939&sr=1-1",
    rating: 7.9,
    src: "img/book_covers/the_power_of_habit329x499.jpg",
    srcSm: "img/book_covers/the_power_of_habit80x121.jpg",
    srcMd: "img/book_covers/the_power_of_habit120x182.jpg",
    srcLg: "img/book_covers/the_power_of_habit200x303.jpg",
    srcXlg: "img/book_covers/the_power_of_habit250x379.jpg",
    background:
      "linear-gradient(185deg, rgb(0, 255, 255), rgb(255, 255, 255), rgb(0, 255, 255));"
  }
];
export default library;
