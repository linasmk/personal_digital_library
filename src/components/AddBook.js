import React, { Component } from "react";
import { FaPlus } from "react-icons/fa";
import AddBookForm from "./AddBookForm";

export default class AddBook extends Component {
  state = {
    formDisplay: undefined,
  };
  handleAddBookFormPopUp = () => {
    this.setState(() => ({
      formDisplay: true,
    }));
  };
  render() {
    return (
      <section className="addbook__wraper">
        <button
          className="button__addbook"
          onClick={this.handleAddBookFormPopUp.bind(this)}
        >
          <FaPlus className="button__addbook_icon" />
        </button>
        <AddBookForm formDisplay={this.state.formDisplay} />
      </section>
    );
  }
}
