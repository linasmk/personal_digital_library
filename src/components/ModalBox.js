/* ========== Dependencies ============ */
import React from "react";
import { FiEdit } from "react-icons/fi";
import { AiOutlineClose, AiOutlineCheckSquare } from "react-icons/ai";
/* ========== Redux ============= */
import { connect } from "react-redux";
import { editBook } from "../store/actionCreators";
/* ========== Components ============ */
import BookPicture from "./BookPicture";
import Modal from "react-modal";
import StarRating from "./StarRating";
import { method } from "lodash";

/* ========== Code ============ */
Modal.setAppElement("#app");

class ModalBox extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      id: props.book ? props.book.id : "",
      title: props.book ? props.book.title : "",
      author: props.book ? props.book.author : "",
      description: props.book ? props.description : "",
      publicationYear: props.book ? props.book.publicationYear : "",
      publisher: props.book ? props.book.publisher : "",
      publicationLength: props.book ? props.book.publicationLength : "",
      category: props.book ? props.book.category : "",
      rating: props.book ? props.book.rating : "",
      src: props.book ? props.book.src : "",
      isContentInEditMode: false,
    };
  }

  toggleEditMode = () => {
    this.setState((prevState) => ({
      isContentInEditMode: !prevState.isContentInEditMode,
    }));
  };
  dispatchEditBook = () => {
    this.setState((book) => {
      this.props.dispatch(editBook(this.state.id, book));
    }),
      this.toggleEditMode();
  };
  onTitleChange = (e) => {
    const title = e.target.textContent;
    this.setState(() => ({
      title,
    }));
  };
  onAuthorChange = (e) => {
    const author = e.target.textContent;
    this.setState(() => ({
      author,
    }));
  };
  onDescriptionChange = (e) => {
    const description = e.target.textContent;
    this.setState(() => ({
      description,
    }));
  };
  onPublicationYearChange = (e) => {
    const publicationYear = e.target.textContent;
    this.setState(() => ({
      publicationYear,
    }));
  };
  onPublisherChange = (e) => {
    const publisher = e.target.textContent;
    this.setState(() => ({
      publisher,
    }));
  };
  onPublicationLengthChange = (e) => {
    const publicationLength = e.target.textContent;
    this.setState(() => ({
      publicationLength,
    }));
  };
  onCategoryChange = (e) => {
    const category = e.target.textContent;
    this.setState(() => ({
      category,
    }));
  };

  render() {
    return (
      <Modal
        isOpen={this.props.viewBookModal}
        onRequestClose={this.props.clearViewBookModal}
        contentLabel={this.props.title}
        closeTimeoutMS={800}
        className="modal"
        key={this.state.id}
      >
        <div className="top-bar h-top-bar-modal">
          {this.state.isContentInEditMode ? (
            <AiOutlineCheckSquare
              className="h-icon-styles"
              viewBox="100 0 900 980"
              /*viewBox attribute is a list of four numbers: 
            min-x, min-y, width and height */
              onClick={this.dispatchEditBook}
            />
          ) : (
            <FiEdit
              className="card__btn-edit h-icon-styles"
              onClick={this.toggleEditMode}
            />
          )}

          <AiOutlineClose
            className="card__btn-close h-icon-styles"
            viewBox="100 50 800 880"
            onClick={this.props.clearViewBookModal}
          />
        </div>
        <h2
          className="modal__title"
          contentEditable={this.state.isContentInEditMode}
          suppressContentEditableWarning
          onBlur={this.onTitleChange}
        >
          {this.state.title}
        </h2>
        <h3 className="modal__author">
          <span>by </span>
          <span
            contentEditable={this.state.isContentInEditMode}
            suppressContentEditableWarning
            onBlur={this.onAuthorChange}
          >
            {this.state.author}
          </span>
        </h3>
        <section className="modal__details_block">
          <section className="details__block_description">
            <h3 className="description__heading">Description:</h3>
            <p
              className="description__p"
              contentEditable={this.state.isContentInEditMode}
              suppressContentEditableWarning
              onBlur={this.onDescriptionChange}
            >
              {this.state.description}
            </p>
          </section>

          <a
            className="details__block_link"
            href={this.state.url}
            target="_blank"
          >
            <BookPicture
              src={this.state.src}
              srcSm={this.state.srcSm}
              srcMd={this.state.srcMd}
              srcLg={this.state.srcLg}
              srcXlg={this.state.srcXlg}
            />
          </a>
          <div className="details__block_table-wrapper">
            <table className="modal__table">
              <tbody>
                <tr>
                  <td>Publication Year</td>
                  <td
                    contentEditable={this.state.isContentInEditMode}
                    suppressContentEditableWarning
                    onBlur={this.onPublicationYearChange}
                  >
                    <strong>{this.state.publicationYear}</strong>
                  </td>
                </tr>
                <tr>
                  <td>Publisher</td>
                  <td
                    contentEditable={this.state.isContentInEditMode}
                    suppressContentEditableWarning
                    onBlur={this.onPublisherChange}
                  >
                    <strong>{this.state.publisher}</strong>
                  </td>
                </tr>
                <tr>
                  <td>Publication Length</td>
                  <td>
                    <strong>
                      <span
                        contentEditable={this.state.isContentInEditMode}
                        suppressContentEditableWarning
                        onBlur={this.onPublicationLengthChange}
                      >
                        {this.state.publicationLength}
                      </span>
                      &nbsp;pages
                    </strong>
                  </td>
                </tr>
                <tr>
                  <td>Category</td>
                  <td
                    contentEditable={this.state.isContentInEditMode}
                    suppressContentEditableWarning
                    onBlur={this.onCategoryChange}
                  >
                    <strong>{this.state.category}</strong>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <StarRating
            // handleStarRating={this.handleStarRating}
            bookRating={this.state.rating}
            bookId={this.state.id}
            starRatingUl="star-rating__ul--modal"
            star="star--modal"
            selected="selected"
            stroke="rgb(255, 152, 0)"
            strokeWidth={1}
          />
        </section>

        <div className="modal__button-bottom_wrapper">
          <button
            className="modal__button-bottom"
            onClick={this.props.clearViewBookModal}
          >
            Okay
          </button>
        </div>
      </Modal>
    );
  }
}

const mapStateToProps = (state, props) => {
  return {
    book: state.books.find((book) => book.id === props.id),
  };
};

export default connect(mapStateToProps)(ModalBox);

// /* ========== Code ============ */

// Modal.setAppElement("#app");

// const ModalBox = ({
//   //Handlers
//   editBookInfo,
//   viewBookModal,
//   clearViewBookModal,
//   //Props
//   book,
//   id,
//   title,
//   author,
//   description,
//   publicationYear,
//   publisher,
//   publicationLength,
//   category,
//   rating,
//   url,
//   imgLink,
//   src,
//   srcSm,
//   srcMd,
//   srcLg,
//   srcXlg,
// }) => {
//   // State
//   const [isContentInEditMode, setEditMode] = useState(false);
//   const [bookTitle, setTitle] = useState(title ? title : "");
//   const [bookAuthor, setAuthor] = useState(author ? author : "");
//   const [bookDescription, setDescription] = useState(
//     description ? description : ""
//   );
//   const [bookPublicationYear, setPublicationYear] = useState(
//     publicationYear ? publicationYear : ""
//   );
//   const [bookPublisher, setPublisher] = useState(publisher ? publisher : "");
//   const [bookPublicationLength, setPublicationLength] = useState(
//     publicationLength ? publicationLength : ""
//   );
//   const [bookCategory, setCategory] = useState(category ? category : "");

//   // Event handlers
//   const editBook =
//   const handleEditMode = () => setEditMode(!isContentInEditMode);
//   const onTitleChange = (e) => setTitle(e.target.textContent);
//   const onAuthorChange = (e) => setAuthor(e.target.textContent);
//   const onDescriptionChange = (e) => setDescription(e.target.textContent);
//   const onPublicationYearChange = (e) =>
//     setPublicationYear(e.target.textContent);
//   const onPublisherChange = (e) => setPublisher(e.target.textContent);
//   const onPublicationLengthChange = (e) =>
//     setPublicationLength(e.target.textContent);
//   const onCategoryChange = (e) => setCategory(e.target.textContent);

//   return (
//     <Modal
//       isOpen={true /*viewBookModal*/}
//       onRequestClose={clearViewBookModal}
//       contentLabel={title}
//       closeTimeoutMS={800}
//       className="modal"
//     >
//       <div className="top-bar h-top-bar-modal">
//         {isContentInEditMode ? (
//           <AiOutlineCheckSquare
//             className="h-icon-styles"
//             viewBox="100 0 900 980"
//             /*viewBox attribute is a list of four numbers:
//             min-x, min-y, width and height */
//             onClick={handleEditMode}
//           />
//         ) : (
//             <FiEdit
//               className="card__btn-edit h-icon-styles"
//               onClick={handleEditMode}
//             />
//           )}

//         <AiOutlineClose
//           className="card__btn-close h-icon-styles"
//           viewBox="100 50 800 880"
//           onClick={clearViewBookModal}
//         />
//       </div>
//       <h2
//         className="modal__title"
//         contentEditable={isContentInEditMode}
//         suppressContentEditableWarning
//         onBlur={onTitleChange}
//       >
//         {title}
//       </h2>
//       <h3 className="modal__author">
//         <span>by </span>
//         <span
//           contentEditable={isContentInEditMode}
//           suppressContentEditableWarning
//           onBlur={onAuthorChange}
//         >
//           {author}
//         </span>
//       </h3>
//       <section className="modal__details_block">
//         <section className="details__block_description">
//           <h3 className="description__heading">Description:</h3>
//           <p
//             className="description__p"
//             contentEditable={isContentInEditMode}
//             suppressContentEditableWarning
//             onBlur={onDescriptionChange}
//           >
//             {description}
//           </p>
//         </section>

//         <a className="details__block_link" href={url} target="_blank">
//           <BookPicture
//             src={src}
//             srcSm={srcSm}
//             srcMd={srcMd}
//             srcLg={srcLg}
//             srcXlg={srcXlg}
//           />
//         </a>
//         <div className="details__block_table-wrapper">
//           <table className="modal__table">
//             <tbody>
//               <tr>
//                 <td>Publication Year</td>
//                 <td
//                   contentEditable={isContentInEditMode}
//                   suppressContentEditableWarning
//                   onBlur={onPublicationYearChange}
//                 >
//                   <strong>{publicationYear}</strong>
//                 </td>
//               </tr>
//               <tr>
//                 <td>Publisher</td>
//                 <td
//                   contentEditable={isContentInEditMode}
//                   suppressContentEditableWarning
//                   onBlur={onPublisherChange}
//                 >
//                   <strong>{publisher}</strong>
//                 </td>
//               </tr>
//               <tr>
//                 <td>Publication Length</td>
//                 <td>
//                   <strong>
//                     <span
//                       contentEditable={isContentInEditMode}
//                       suppressContentEditableWarning
//                       onBlur={onPublicationLengthChange}
//                     >
//                       {publicationLength}
//                     </span>
//                     &nbsp;pages
//                   </strong>
//                 </td>
//               </tr>
//               <tr>
//                 <td>Category</td>
//                 <td
//                   contentEditable={isContentInEditMode}
//                   suppressContentEditableWarning
//                   onBlur={onCategoryChange}
//                 >
//                   <strong>{category}</strong>
//                 </td>
//               </tr>
//             </tbody>
//           </table>
//         </div>
//         <StarRatingModal />
//       </section>

//       <div className="modal__button-bottom_wrapper">
//         <button className="modal__button-bottom" onClick={clearViewBookModal}>
//           Okay
//         </button>
//       </div>
//     </Modal>
//   );
// };
// const mapStateToProps = (state, { id }) => {
//   return {
//     book: state.books.find(
//       (book) => book.id === id
//     ),
//   };
// };

// export default connect(mapStateToProps)(ModalBox);
