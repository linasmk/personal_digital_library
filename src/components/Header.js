/* ========= App Dependencies ============= */
import React from "react";
/* ========= Components ============= */
import BooksSummary from "./BooksSummary";
/* ========= Code ============= */

const Header = (props) => {
  return (
    <header>
      <h1 className="header__main_heading">{props.title}</h1>
      <BooksSummary />
    </header>
  );
};
Header.defaultProps = {
  title: "Personal Digital Library",
};
export default Header;
