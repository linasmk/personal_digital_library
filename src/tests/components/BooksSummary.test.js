/* ========= App Dependencies ============= */
import React from "react";
import { shallow } from "enzyme";
/* ========== Components ========== */
import { BooksSummary } from "../../components/BooksSummary";

it("renders BooksSummary correctly with 1 book", () => {
  const wrapper = shallow(<BooksSummary booksCount={1} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders booksSummary correctly with several books", () => {
  const wrapper = shallow(<BooksSummary booksCount={5} />);
  expect(wrapper).toMatchSnapshot();
});
