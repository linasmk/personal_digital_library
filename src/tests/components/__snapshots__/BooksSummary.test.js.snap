// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BooksSummary correctly with 1 book 1`] = `
<section
  className="header__summary_block"
>
  <p
    className="header__summary_p"
  >
    Filtered books:
     
    <span
      className="header__summary_output"
    >
      1
    </span>
  </p>
</section>
`;

exports[`renders booksSummary correctly with several books 1`] = `
<section
  className="header__summary_block"
>
  <p
    className="header__summary_p"
  >
    Filtered books:
     
    <span
      className="header__summary_output"
    >
      5
    </span>
  </p>
</section>
`;
